many-fuzzy/src/main.rs

```diff
@@ -0,0 +1,348 @@
+    rng: &mut impl rand::Rng,
+    count: u64,
+    builder: RequestMessageBuilder,
+    message: &str,
+) -> Result<Vec<RequestMessage>, String> {
+    let re = regex::Regex::new(r"%%|%\(([^\)]*)\)").unwrap();
+    let mut messages = Vec::new();
+
+    for _ in 0..count {
+        let mut builder = builder.clone();
+        let data = re.replace(message, |cap: &regex::Captures| {
+            let ty = cap.get(1).unwrap();
+            let mut generator = parsers::fuzz_string::generator(ty.as_str())
+                .expect("Could not parse fuzzy parameters");
+            let value = generator.fuzz(rng);
+
+            value
+        });
+        builder.data(cbor_diag::parse_diag(&data).unwrap().to_bytes());
+        messages.push(builder.build().map_err(|e| e.to_string())?);
+    }
+
+    Ok(messages)
+}
+
+/// Start a thread that waits every 2 seconds and show the current progress.
+fn start_counting_thread(statistics: Statistics) -> (Sender<()>, JoinHandle<()>) {
+    let (stop_thread_tx, mut stop_thread_rx) = tokio::sync::oneshot::channel();
+    let thread_handle = tokio::spawn(async move {
+        let mut interval = interval(Duration::from_millis(1000));
+        // Only show every 5 seconds if nothing changed.
+        let mut last_response = 0;
+        let mut last_request = 0;
+        let mut last_counter = 1;
+
+        loop {
+            tokio::select! {
+                _ = interval.tick() => {
+                    let (request, response) = (
+                        statistics.request_counter.load(),
+                        statistics.response_counter.load()
+                    );
+
+                    if last_response == response && last_request == request {
+                        last_counter -= 1;
+                        if last_counter == 0 {
+                            tracing::info!(
+                                "Requests: {:5} sent, {:5} received",
+                                request, response
+                            );
+                            last_counter = 5;
+                        }
+                    } else {
+                        last_response = response;
+                        last_request = request;
+                        last_counter = 5;
+                        tracing::info!(
+                            "Requests: {:5} sent, {:5} received",
+                            request, response
+                        );
+                    }
+                },
+                _ = &mut stop_thread_rx => {
+                    tracing::info!(
+                        "Requests: {:5} sent, {:5} received",
+                        statistics.request_counter.load(),
+                        statistics.response_counter.load()
+                    );
+                    break;
+                }
+            }
+        }
+    });
+
+    (stop_thread_tx, thread_handle)
+}
+
+/// Print out the statistics results to the user.
+async fn show_statistics(statistics: Statistics) {
```

Comment:
You could probably use something to print pretty tables like [term-table](https://crates.io/crates/term-table) or other.

Reply:
Maybe another time :)